scripts/tools/InitializeQueue.php

```diff
@@ -188,8 +160,50 @@ private function checkParams(array $params)
             }
         }
 
-        if ($this->wantedBroker == self::BROKER_RDS && !$this->persistenceId) {
+        $this->validateBrokersWithPersistence();
+    }
+
+    private function validateBrokersWithPersistence(): void
+    {
+        if (in_array($this->wantedBroker, [BrokerFactory::BROKER_RDS, BrokerFactory::BROKER_NEW_SQL], true)
+            && !$this->persistenceId) {
             throw new \InvalidArgumentException('Persistence id (--persistence=...) needs to be set for RDS.');
         }
     }
+
+    private function registerBroker(array $params, QueueDispatcherInterface $queueService): bool
+    {
+        $reRegister = false;
+        // if any new change is wanted on queues
+        if (count($params) > 0) {
+            // BROKER settings
+            if ($this->wantedBroker) {
+                $brokerFactory = $this->getBrokerFactory();
+                $broker = $brokerFactory->create($this->wantedBroker, $this->persistenceId, $this->receive ?: 1);
+
+                if (!is_null($this->queue)) {
+                    $queue = $queueService->getQueue($this->queue);
+                    $queue->setBroker(clone $broker);
+                } else {
+                    foreach ($queueService->getQueues() as $queue) {
+                        $queue->setBroker(clone $broker);
+                    }
+                }
+            }
+
+            // STRATEGY settings
+            if ($this->strategy) {
+                $queueService->setTaskSelector($this->strategy);
+            }
+
+            $reRegister = true;
+        }
+
+        return $reRegister;
+    }
```

Comment:
The only reason this method return boolean is because you need to log it and register, so why not having this here?

```
$this->registerService(QueueDispatcherInterface::SERVICE_ID, $queueService);

$report->add(Report::createSuccess('Queue service re-registered.'));
```
You can simplify / reduce the method complexity by:

```suggestion
    private function registerBroker(array $params, Report $report, QueueDispatcherInterface $queueService): void
    {
        if (count($params) === 0) {
            return;
        }

        if ($this->strategy) {
            $queueService->setTaskSelector($this->strategy);
        }

        if (!$this->wantedBroker) {
            return;
        }

        $broker = $this->getBrokerFactory()
            ->create($this->wantedBroker, $this->persistenceId, $this->receive ?: 1);

        $queues = is_null($this->queue) ? $queueService->getQueues() : [$queueService->getQueue($this->queue)]

        foreach ($queueService->getQueues() as $queue) {
            $queue->setBroker(clone $broker);
        }
        
        $this->registerService(QueueDispatcherInterface::SERVICE_ID, $queueService);

        $report->add(Report::createSuccess('Queue service re-registered.'));
    }
```